io_scene_niftools/modules/nif_export/geometry/mesh/__init__.py

```diff
@@ -562,21 +585,45 @@ def select_unweighted_vertices(self, unweighted_vertices):
             # select unweighted vertices
             bpy.ops.mesh.select_ungrouped(extend=False)
 
-            raise io_scene_niftools.utils.logging.NifError("Cannot export mesh with unweighted vertices. "
-                                     "The unweighted vertices have been selected in the mesh so they can easily be identified.")
-
+            raise NifError("Cannot export mesh with unweighted vertices. "
+                           "The unweighted vertices have been selected in the mesh so they can easily be identified.")
 
     def select_unassigned_polygons(self, b_mesh, b_obj, polygons_without_bodypart):
         """Select any faces which are not weighted to a vertex group"""
+        ngon_mesh = b_obj.data
+        # make vertex: poly map of the untriangulated mesh
+        vert_poly_dict = dict(zip(range(len(ngon_mesh.vertices)), [set() for i in range(len(ngon_mesh.vertices))]))
```

Comment:
`len(ngon_mesh.vertices)` used twice so would pull out to `num_vertices` but will also improve readability.

Reply:
Simplify this to vert_poly_dict = {i: set() for for i in range(len(ngon_mesh.vertices))}